TrackerPatterns plugin does not respect proxy directives in settings.yml

<!-- PLEASE FILL THESE FIELDS, IT REALLY HELPS THE MAINTAINERS OF SearXNG -->

**Version of SearXNG, commit number if you are using on master branch and stipulate if you forked SearXNG**: searxng-docker (2025.7.18+ff2e0ea)

**How did you install SearXNG?**: using searxng-docker

**What happened?**
<!-- A clear and concise description of what the bug is. -->
Following https://github.com/searxng/searxng/pull/4678, an HTTP call is made at the startup of searxng in order to fetch a couple of lists of tracker patterns

https://github.com/searxng/searxng/blob/ff2e0ea2788a04ae5a13fc90b3725828a1ebc026/searx/data/tracker_patterns.py#L74

This HTTP call does not seem to go through the proxy configured in `settings.yml`.

If searxng is not able to perform this HTTP call (either due to a network error, or in my case because connections attempts not made through the proxy are blocked), it crashes during startup.

**How To Reproduce**
<!-- How can we reproduce this issue? (as minimally and as precisely as possible) -->

Configure searxng to use a proxy and block all other connection attempts using a firewall.

Relevant configuration element (`tor:9050` is hosted in another docker container):

```yml
use_default_settings: true
outgoing:
  using_tor_proxy: true
  proxies:
    all://:
      - socks5h://tor:9050
```

**Expected behavior**
Either:
- No tracker list are downloaded, allowing searxng to run without this specific plugin (in this case, the error should be handled instead of crashing searxng)
- Tracker lists are downloaded through the proxy configured in settings.yml

**Screenshots & Logs**
```
SearXNG 2025.7.18+ff2e0ea
[INFO] Starting granian (main PID: 1)
[INFO] Listening at: http://:::8080
[INFO] Spawning worker-1 with PID: 11
Process granian-worker:
Traceback (most recent call last):
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/httpx/_transports/default.py", line 101, in map_httpcore_exceptions
    yield
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/httpx/_transports/default.py", line 250, in handle_request
    resp = self._pool.handle_request(req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/httpcore/_sync/connection_pool.py", line 256, in handle_request
    raise exc from None
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/httpcore/_sync/connection_pool.py", line 236, in handle_request
    response = connection.handle_request(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/httpcore/_sync/connection.py", line 101, in handle_request
    raise exc
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/httpcore/_sync/connection.py", line 78, in handle_request
    stream = self._connect(request)
             ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/httpcore/_sync/connection.py", line 124, in _connect
    stream = self._network_backend.connect_tcp(**kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/httpcore/_backends/sync.py", line 207, in connect_tcp
    with map_exceptions(exc_map):
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/contextlib.py", line 158, in __exit__
    self.gen.throw(value)
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/httpcore/_exceptions.py", line 14, in map_exceptions
    raise to_exc(exc) from exc
httpcore.ConnectError: [Errno 99] Address not available

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3.12/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/usr/lib/python3.12/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/granian/server/mp.py", line 64, in wrapped
    callback = callback_loader()
               ^^^^^^^^^^^^^^^^^
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/granian/_internal.py", line 61, in load_target
    module = load_module(path)
             ^^^^^^^^^^^^^^^^^
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/granian/_internal.py", line 43, in load_module
    __import__(module_name)
  File "/usr/local/searxng/searx/webapp.py", line 1429, in <module>
    init()
  File "/usr/local/searxng/searx/webapp.py", line 1418, in init
    searx.plugins.initialize(app)
  File "/usr/local/searxng/searx/plugins/__init__.py", line 110, in initialize
    STORAGE.init(app)
  File "/usr/local/searxng/searx/plugins/_core.py", line 248, in init
    if not plg.init(app):
           ^^^^^^^^^^^^^
  File "/usr/local/searxng/searx/plugins/tracker_url_remover.py", line 42, in init
    TRACKER_PATTERNS.init()
  File "/usr/local/searxng/searx/data/tracker_patterns.py", line 45, in init
    self.load()
  File "/usr/local/searxng/searx/data/tracker_patterns.py", line 52, in load
    for rule in self.iter_clear_list():
                ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/searxng/searx/data/tracker_patterns.py", line 74, in iter_clear_list
    resp = httpx.get(url, timeout=3)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/httpx/_api.py", line 195, in get
    return request(
           ^^^^^^^^
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/httpx/_api.py", line 109, in request
    return client.request(
           ^^^^^^^^^^^^^^^
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/httpx/_client.py", line 825, in request
    return self.send(request, auth=auth, follow_redirects=follow_redirects)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/httpx/_client.py", line 914, in send
    response = self._send_handling_auth(
               ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/httpx/_client.py", line 942, in _send_handling_auth
    response = self._send_handling_redirects(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/httpx/_client.py", line 979, in _send_handling_redirects
    response = self._send_single_request(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/httpx/_client.py", line 1014, in _send_single_request
    response = transport.handle_request(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/httpx/_transports/default.py", line 249, in handle_request
    with map_httpcore_exceptions():
         ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/contextlib.py", line 158, in __exit__
    self.gen.throw(value)
  File "/usr/local/searxng/venv/lib/python3.12/site-packages/httpx/_transports/default.py", line 118, in map_httpcore_exceptions
    raise mapped_exc(message) from exc
httpx.ConnectError: [Errno 99] Address not available
[ERROR] Unexpected exit from worker-1
[INFO] Shutting down granian
```

**Additional context**

--- a/searx/data/tracker_patterns.py
+++ b/searx/data/tracker_patterns.py
@@ -10,9 +10,8 @@ import re
 from collections.abc import Iterator
 from urllib.parse import urlparse, urlunparse, parse_qsl, urlencode
 
-import httpx
-
 from searx.data.core import get_cache, log
+from searx.network import get as http_get
 
 RuleType = tuple[str, list[str], list[str]]
 
@@ -44,7 +43,7 @@ class TrackerPatternsDB:
             self.cache.properties.set("tracker_patterns loaded", "OK")
             self.load()
         # F I X M E:
-        #     do we need a maintenance .. rember: database is stored
+        #     do we need a maintenance .. remember: database is stored
         #     in /tmp and will be rebuild during the reboot anyway
 
     def load(self):
@@ -71,7 +70,7 @@ class TrackerPatternsDB:
     def iter_clear_list(self) -> Iterator[RuleType]:
         resp = None
         for url in self.CLEAR_LIST_URL:
-            resp = httpx.get(url, timeout=3)
+            resp = http_get(url, timeout=3)
             if resp.status_code == 200:
                 break
             log.warning(f"TRACKER_PATTERNS: ClearURL ignore HTTP {resp.status_code} {url}")

--- a/searxng_extra/update/update_external_bangs.py
+++ b/searxng_extra/update/update_external_bangs.py
@@ -8,10 +8,10 @@ from :py:obj:`BANGS_URL`.
 """
 
 import json
-import httpx
 
 from searx.external_bang import LEAF_KEY
 from searx.data import data_dir
+from searx.network import get as http_get
 
 DATA_FILE = data_dir / 'external_bangs.json'
 
@@ -24,7 +24,7 @@ HTTP_COLON = 'http:'
 
 def main():
     print(f'fetch bangs from {BANGS_URL}')
-    response = httpx.get(BANGS_URL)
+    response = http_get(BANGS_URL)
     response.raise_for_status()
     ddg_bangs = json.loads(response.content.decode())
     trie = parse_ddg_bangs(ddg_bangs)

